data/dataset.py

```diff
@@ -15,85 +14,50 @@
 class TokenizedMidiDataset:
     def __init__(
         self,
-        split: str = "train",
+        src_tokenizer: Tokenizer,
+        tgt_tokenizer: Tokenizer,
+        dataset: Dataset,
         n_dstart_bins: int = 3,
         n_duration_bins: int = 3,
         n_velocity_bins: int = 3,
         sequence_len: int = 128,
-        device: str = "cpu",
     ):
-        self.split = split
         self.sequence_len = sequence_len
-        self.device = device
+        self.tokenizer_src = src_tokenizer
+        self.tokenizer_tgt = tgt_tokenizer
+
+        self.dataset = dataset
 
         self.quantizer = MidiQuantizer(
             n_dstart_bins=n_dstart_bins,
             n_duration_bins=n_duration_bins,
             n_velocity_bins=n_velocity_bins,
         )
 
-        self.tokenizer_src = Tokenizer(keys=["pitch", "dstart_bin", "duration_bin", "velocity_bin"])
-        self.tokenizer_tgt = VelocityTokenizer()
-
-        self.dataset = load_dataset(path="roszcz/maestro-v1", split=split)
+        self.src_vocab, self.tgt_vocab = self.build_vocab()
 
-        self.src_vocab, self.tgt_vocab = self._build_vocab()
-
-        self.processed_records, self.unprocessed_records = self.load_dataset()
+        self.processed_records, self.unprocessed_records = self._build_dataset()
```

Comment:
```suggestion
        self.records = self._build_dataset()
```

My intuition is that we don't need this processed/unprocessed split here, and we should be good with a single set of records holding both processed and unprocessed values (i.e. `[pitch, duration, duration_bin, ...]`)

Right now it's a bit confusing, and for example code like this:

```python
src_tokens = self.tokenizer_src(processed_record)
tgt_tokens = self.tokenizer_tgt(unprocessed_record)
```
is very enigmatic, and I think this would be better:

```python
# All MIDI in low resolution
src_tokens = self.tokenizer_src(record)

# High resolution velocity
tgt_tokens = self.tokenizer_tgt(record)
```

I'm guessing that you were following the language translation design, and processed/unporcessed are equivalent of english and german, but I think it doesn't fit our use case great :)

Reply:
done

Reply:
done :)